src/k8s/pkg/k8sd/app/hooks_join.go

```diff
@@ -158,10 +158,16 @@ func (a *App) onPostJoin(ctx context.Context, s state.State, initConfig map[stri
 		}
 		cluster := make([]string, len(members))
 		for _, member := range members {
-			cluster = append(cluster, fmt.Sprintf("%s:%d", member.Address.Addr(), cfg.Datastore.GetK8sDqlitePort()))
+			var address string
+			if member.Address.Addr().Is6() {
+				address = fmt.Sprintf("[%s]", member.Address.Addr().String())
+			} else {
+				address = member.Address.Addr().String()
+			}
+			cluster = append(cluster, fmt.Sprintf("%s:%d", address, cfg.Datastore.GetK8sDqlitePort()))
 		}
 
-		address := fmt.Sprintf("%s:%d", nodeIP.String(), cfg.Datastore.GetK8sDqlitePort())
+		address := fmt.Sprintf("%s:%d", utils.ToIPString(nodeIP), cfg.Datastore.GetK8sDqlitePort())
```

Comment:
We need to pass this `address` on line 195 for kubeapiserver setup as well currently it passes nodeIP. This is needed for setting the advertise-address flag.

Reply:
This address is the host:port of the k8s-dqlite endpoint while the kube-apiserver setup expects the IP of the apiserver. Or do I misunderstand your question?
Properly stringifying this IP is done now in the function itself.

Reply:
Does that string conversion work for ipv6 in kube-apiserver to pass to the advertise-address flag is my concern.

Reply:
For the record: It does not.
Kube-Apiserver expects the IPv6 to not have the brackets otherwise it fails with:
`Error: invalid argument "[fd42:1e1e:7a2f:326e:216:3eff:fe3c:db54]" for "--advertise-address" flag: failed to parse IP`.
So this was correct by accident but thanks to your comment we have double-checked that :)